provisioning/device/src/tpm_registration.ts

```diff
@@ -189,8 +188,8 @@ export class TpmRegistration extends EventEmitter implements RegistrationClient
     });
   }
 
-  register(registrationId: string, forceRegistration: boolean, callback: (err?: Error, result?: any) => void): void {
-    this._fsm.handle('register', callback);
+  register(request: RegistrationRequest, callback: (err?: Error, result?: any) => void): void {
```

Comment:
>RegistrationRequest [](start = 20, length = 19)

as much as I like the idea of packaging these together internally, I'm not sure other SDKs have such an object in their public API. we need to make sure we present the same set of parameters in the same API calls as much as possible. can you double-check what the others do and make sure we are aligned as far as public APIs are concerned?